### PR TITLE
Feature/default graph

### PR DESCRIPTION
**Pull request by neuneck**
I implemented a default graph in the flowpipe.graph submodule. Ideally, it would go in the `__init__.py`, but that would lead to more restructuring than I am comfortable with for such a qualitiy-of-life change.

The default setting for `graph` in the `INode.__init__()` is now the string 'default', which indicates the node should be added to the default graph. Explicitly giving `None` as a graph argument still prevents the node from being added anywhere. Of course, explicitly giving a graph also just adds the node there.

The default graph can be interacted with using the handler methods
```
get_default_graph()
set_default_graph(graph)
reset_default_graph()
```
which are pretty self-explanatory.

I had to adapt the function wrapper such that the newly created Node class does have an instance added to the default graph right away.

I also had to move the deserialization of nodes and graphs into the utilities submodule in order to prevent a cyclical import. The respective classes' staticmethods point there, in order to keep backwards compatibility.

Finally, I had to add a fixture to clear the default graph before running most tests, as well as explicitly preventing some nodes in some tests from being added to the default graph.

### CODE REVIEW

File: flowpipe/graph.py [hunk 1]
```diff
@@ -191,3 +174,25 @@ def list_repr(self):
         for node in self.evaluation_sequence:
             pretty.append(node.list_repr())
         return '\n '.join(pretty)
+
+
+default_graph = Graph(name="default")
```
[the remark this answers is not in the document]
Comment (neuneck) [reply flowpipe/graph.py]: As far as I know, introducing own dunders is strongly advised against - it's pretty much the unique handle for built-in functionality to attach to. As a result introducing own dunders could conflict with new built-ins introduced later and would be make the code harder to read, as the reader would expect '__default__' to be picked up by some build-in functionality.

**Comment by PaulSchweizer** [comment]
Not sure why the py 3.4 and 3.5 tests are failing, seems like something we introduced in the latest commit? Is it the conftest.py? Doesn't make sense to me though ...

**Comment by neuneck** [comment]
I am also at a loss. The CI logs show that it can't find the module 'unicodedata', which is part of the standard library for 3.4 and 3.5. Does Travis CI reset the environment between steps? I had some trouble with python on a drone CI that had their root in something like that.

**Comment by PaulSchweizer** [comment]
Seems like travis is a  bit flaky in general these past few weeks, had to restart some jobs lately that were failing due to missing packages and so on but were fine after a restart. Restart doesn't help here though. My only wild guess is really the conftest.py file as the issue seems to be connected to the tests somehow

**Comment by neuneck** [comment]
It seems your gut feeling was right - I moved the fixture into the individual test files and now the CI works again.

**Pull request merged**